app.py

```diff
@@ -100,7 +105,15 @@ def on_new_tweet():
         return "weird!"
 
     puzzle_id = parts[0]
```

Comment:
See where the variables "parts" and "text_parts" are above - the goal is to replace those variables, and all of the text splitting and error checking, with a simple multivariable regular expression, doing the validation all in one swoop...

Reply:
See this Stack Overflow post for more info on how one can extract multiple named regex vars: https://stackoverflow.com/questions/1800817/how-can-i-get-part-of-regex-match-as-a-variable-in-python

Reply:
Just updated it!